[Bug]: Outdated instructions for launcher unit test

### Contact Details

_No response_

### What happened?

I found the instructions at https://github.com/llm-d-incubation/llm-d-fast-model-actuation/blob/main/inference_server/launcher/howto.md and tried to follow them, adapting to use a virtual environment. Several things went wrong. The following typescript shows the flailing and eventual success.

```console
mspreitz@mjs13 llm-d-fast-model-actuation % python3 -m venv inference_server/launcher/venv
mspreitz@mjs13 llm-d-fast-model-actuation % bash
bash-5.3$ 
bash-5.3$ cd inference_server/launcher/
bash-5.3$ ls
__init__.py		gputranslator.py	launcher.py		tests
__pycache__		howto.md		requirements.txt	venv
bash-5.3$ . venv/bin/activate
(venv) bash-5.3$ which pip
/Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher/venv/bin/pip
(venv) bash-5.3$ pip install --upgrade pip
Requirement already satisfied: pip in ./venv/lib/python3.14/site-packages (26.0)
Collecting pip
  Downloading pip-26.0.1-py3-none-any.whl.metadata (4.7 kB)
Downloading pip-26.0.1-py3-none-any.whl (1.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 13.3 MB/s  0:00:00
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 26.0
    Uninstalling pip-26.0:
      Successfully uninstalled pip-26.0
Successfully installed pip-26.0.1
(venv) bash-5.3$ pip install -r requirements.txt 
Ignoring vllm: markers 'sys_platform != "darwin" or platform_machine != "arm64"' don't match your environment
Collecting fastapi (from -r requirements.txt (line 1))
  Using cached fastapi-0.135.1-py3-none-any.whl.metadata (30 kB)
Collecting pydantic (from -r requirements.txt (line 2))
  Using cached pydantic-2.12.5-py3-none-any.whl.metadata (90 kB)
Collecting uvicorn (from -r requirements.txt (line 3))
  Using cached uvicorn-0.42.0-py3-none-any.whl.metadata (6.7 kB)
Collecting uvloop (from -r requirements.txt (line 4))
  Using cached uvloop-0.22.1-cp314-cp314-macosx_10_13_universal2.whl.metadata (4.9 kB)
Collecting nvidia-ml-py (from -r requirements.txt (line 5))
  Using cached nvidia_ml_py-13.595.45-py3-none-any.whl.metadata (9.7 kB)
Collecting kubernetes==31.0.0 (from -r requirements.txt (line 6))
  Using cached kubernetes-31.0.0-py2.py3-none-any.whl.metadata (1.5 kB)
Collecting certifi>=14.05.14 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached certifi-2026.2.25-py3-none-any.whl.metadata (2.5 kB)
Collecting six>=1.9.0 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached six-1.17.0-py2.py3-none-any.whl.metadata (1.7 kB)
Collecting python-dateutil>=2.5.3 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pyyaml>=5.4.1 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached pyyaml-6.0.3-cp314-cp314-macosx_11_0_arm64.whl.metadata (2.4 kB)
Collecting google-auth>=1.0.1 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached google_auth-2.49.1-py3-none-any.whl.metadata (6.2 kB)
Collecting websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached websocket_client-1.9.0-py3-none-any.whl.metadata (8.3 kB)
Collecting requests (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached requests-2.32.5-py3-none-any.whl.metadata (4.9 kB)
Collecting requests-oauthlib (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached requests_oauthlib-2.0.0-py2.py3-none-any.whl.metadata (11 kB)
Collecting oauthlib>=3.2.2 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached oauthlib-3.3.1-py3-none-any.whl.metadata (7.9 kB)
Collecting urllib3>=1.24.2 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached urllib3-2.6.3-py3-none-any.whl.metadata (6.9 kB)
Collecting durationpy>=0.7 (from kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached durationpy-0.10-py3-none-any.whl.metadata (340 bytes)
Collecting starlette>=0.46.0 (from fastapi->-r requirements.txt (line 1))
  Using cached starlette-0.52.1-py3-none-any.whl.metadata (6.3 kB)
Collecting typing-extensions>=4.8.0 (from fastapi->-r requirements.txt (line 1))
  Using cached typing_extensions-4.15.0-py3-none-any.whl.metadata (3.3 kB)
Collecting typing-inspection>=0.4.2 (from fastapi->-r requirements.txt (line 1))
  Using cached typing_inspection-0.4.2-py3-none-any.whl.metadata (2.6 kB)
Collecting annotated-doc>=0.0.2 (from fastapi->-r requirements.txt (line 1))
  Using cached annotated_doc-0.0.4-py3-none-any.whl.metadata (6.6 kB)
Collecting annotated-types>=0.6.0 (from pydantic->-r requirements.txt (line 2))
  Using cached annotated_types-0.7.0-py3-none-any.whl.metadata (15 kB)
Collecting pydantic-core==2.41.5 (from pydantic->-r requirements.txt (line 2))
  Using cached pydantic_core-2.41.5-cp314-cp314-macosx_11_0_arm64.whl.metadata (7.3 kB)
Collecting click>=7.0 (from uvicorn->-r requirements.txt (line 3))
  Using cached click-8.3.1-py3-none-any.whl.metadata (2.6 kB)
Collecting h11>=0.8 (from uvicorn->-r requirements.txt (line 3))
  Using cached h11-0.16.0-py3-none-any.whl.metadata (8.3 kB)
Collecting pyasn1-modules>=0.2.1 (from google-auth>=1.0.1->kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached pyasn1_modules-0.4.2-py3-none-any.whl.metadata (3.5 kB)
Collecting cryptography>=38.0.3 (from google-auth>=1.0.1->kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached cryptography-46.0.5-cp311-abi3-macosx_10_9_universal2.whl.metadata (5.7 kB)
Collecting cffi>=2.0.0 (from cryptography>=38.0.3->google-auth>=1.0.1->kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached cffi-2.0.0-cp314-cp314-macosx_11_0_arm64.whl.metadata (2.6 kB)
Collecting pycparser (from cffi>=2.0.0->cryptography>=38.0.3->google-auth>=1.0.1->kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached pycparser-3.0-py3-none-any.whl.metadata (8.2 kB)
Collecting pyasn1<0.7.0,>=0.6.1 (from pyasn1-modules>=0.2.1->google-auth>=1.0.1->kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached pyasn1-0.6.3-py3-none-any.whl.metadata (8.4 kB)
Collecting anyio<5,>=3.6.2 (from starlette>=0.46.0->fastapi->-r requirements.txt (line 1))
  Using cached anyio-4.12.1-py3-none-any.whl.metadata (4.3 kB)
Collecting idna>=2.8 (from anyio<5,>=3.6.2->starlette>=0.46.0->fastapi->-r requirements.txt (line 1))
  Using cached idna-3.11-py3-none-any.whl.metadata (8.4 kB)
Collecting charset_normalizer<4,>=2 (from requests->kubernetes==31.0.0->-r requirements.txt (line 6))
  Using cached charset_normalizer-3.4.6-cp314-cp314-macosx_10_15_universal2.whl.metadata (40 kB)
Using cached kubernetes-31.0.0-py2.py3-none-any.whl (1.9 MB)
Using cached fastapi-0.135.1-py3-none-any.whl (116 kB)
Using cached pydantic-2.12.5-py3-none-any.whl (463 kB)
Using cached pydantic_core-2.41.5-cp314-cp314-macosx_11_0_arm64.whl (1.9 MB)
Using cached uvicorn-0.42.0-py3-none-any.whl (68 kB)
Using cached uvloop-0.22.1-cp314-cp314-macosx_10_13_universal2.whl (1.4 MB)
Using cached nvidia_ml_py-13.595.45-py3-none-any.whl (51 kB)
Using cached annotated_doc-0.0.4-py3-none-any.whl (5.3 kB)
Using cached annotated_types-0.7.0-py3-none-any.whl (13 kB)
Using cached certifi-2026.2.25-py3-none-any.whl (153 kB)
Using cached click-8.3.1-py3-none-any.whl (108 kB)
Using cached durationpy-0.10-py3-none-any.whl (3.9 kB)
Using cached google_auth-2.49.1-py3-none-any.whl (240 kB)
Using cached cryptography-46.0.5-cp311-abi3-macosx_10_9_universal2.whl (7.2 MB)
Using cached cffi-2.0.0-cp314-cp314-macosx_11_0_arm64.whl (181 kB)
Using cached h11-0.16.0-py3-none-any.whl (37 kB)
Using cached oauthlib-3.3.1-py3-none-any.whl (160 kB)
Using cached pyasn1_modules-0.4.2-py3-none-any.whl (181 kB)
Using cached pyasn1-0.6.3-py3-none-any.whl (83 kB)
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Using cached pyyaml-6.0.3-cp314-cp314-macosx_11_0_arm64.whl (173 kB)
Using cached six-1.17.0-py2.py3-none-any.whl (11 kB)
Using cached starlette-0.52.1-py3-none-any.whl (74 kB)
Using cached anyio-4.12.1-py3-none-any.whl (113 kB)
Using cached idna-3.11-py3-none-any.whl (71 kB)
Using cached typing_extensions-4.15.0-py3-none-any.whl (44 kB)
Using cached typing_inspection-0.4.2-py3-none-any.whl (14 kB)
Using cached urllib3-2.6.3-py3-none-any.whl (131 kB)
Using cached websocket_client-1.9.0-py3-none-any.whl (82 kB)
Using cached pycparser-3.0-py3-none-any.whl (48 kB)
Using cached requests-2.32.5-py3-none-any.whl (64 kB)
Using cached charset_normalizer-3.4.6-cp314-cp314-macosx_10_15_universal2.whl (294 kB)
Using cached requests_oauthlib-2.0.0-py2.py3-none-any.whl (24 kB)
Installing collected packages: nvidia-ml-py, durationpy, websocket-client, uvloop, urllib3, typing-extensions, six, pyyaml, pycparser, pyasn1, oauthlib, idna, h11, click, charset_normalizer, certifi, annotated-types, annotated-doc, uvicorn, typing-inspection, requests, python-dateutil, pydantic-core, pyasn1-modules, cffi, anyio, starlette, requests-oauthlib, pydantic, cryptography, google-auth, fastapi, kubernetes
Successfully installed annotated-doc-0.0.4 annotated-types-0.7.0 anyio-4.12.1 certifi-2026.2.25 cffi-2.0.0 charset_normalizer-3.4.6 click-8.3.1 cryptography-46.0.5 durationpy-0.10 fastapi-0.135.1 google-auth-2.49.1 h11-0.16.0 idna-3.11 kubernetes-31.0.0 nvidia-ml-py-13.595.45 oauthlib-3.3.1 pyasn1-0.6.3 pyasn1-modules-0.4.2 pycparser-3.0 pydantic-2.12.5 pydantic-core-2.41.5 python-dateutil-2.9.0.post0 pyyaml-6.0.3 requests-2.32.5 requests-oauthlib-2.0.0 six-1.17.0 starlette-0.52.1 typing-extensions-4.15.0 typing-inspection-0.4.2 urllib3-2.6.3 uvicorn-0.42.0 uvloop-0.22.1 websocket-client-1.9.0
(venv) bash-5.3$ 
(venv) bash-5.3$ python -m pytest test_launcher.py -v
/Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher/venv/bin/python: No module named pytest
(venv) bash-5.3$ 
(venv) bash-5.3$ pip install pytest
Collecting pytest
  Using cached pytest-9.0.2-py3-none-any.whl.metadata (7.6 kB)
Collecting iniconfig>=1.0.1 (from pytest)
  Using cached iniconfig-2.3.0-py3-none-any.whl.metadata (2.5 kB)
Collecting packaging>=22 (from pytest)
  Using cached packaging-26.0-py3-none-any.whl.metadata (3.3 kB)
Collecting pluggy<2,>=1.5 (from pytest)
  Using cached pluggy-1.6.0-py3-none-any.whl.metadata (4.8 kB)
Collecting pygments>=2.7.2 (from pytest)
  Using cached pygments-2.19.2-py3-none-any.whl.metadata (2.5 kB)
Using cached pytest-9.0.2-py3-none-any.whl (374 kB)
Using cached pluggy-1.6.0-py3-none-any.whl (20 kB)
Using cached iniconfig-2.3.0-py3-none-any.whl (7.5 kB)
Using cached packaging-26.0-py3-none-any.whl (74 kB)
Using cached pygments-2.19.2-py3-none-any.whl (1.2 MB)
Installing collected packages: pygments, pluggy, packaging, iniconfig, pytest
Successfully installed iniconfig-2.3.0 packaging-26.0 pluggy-1.6.0 pygments-2.19.2 pytest-9.0.2
(venv) bash-5.3$ 
(venv) bash-5.3$ python -m pytest test_launcher.py -v
======================================================= test session starts =======================================================
platform darwin -- Python 3.14.3, pytest-9.0.2, pluggy-1.6.0 -- /Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher
plugins: anyio-4.12.1
collected 0 items                                                                                                                 

====================================================== no tests ran in 0.00s ======================================================
ERROR: file or directory not found: test_launcher.py

(venv) bash-5.3$ ls
__init__.py		gputranslator.py	launcher.py		tests
__pycache__		howto.md		requirements.txt	venv
(venv) bash-5.3$ 
(venv) bash-5.3$ 
(venv) bash-5.3$ python -m pytest tests/test_launcher.py -v
======================================================= test session starts =======================================================
platform darwin -- Python 3.14.3, pytest-9.0.2, pluggy-1.6.0 -- /Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher
plugins: anyio-4.12.1
collected 0 items / 1 error                                                                                                       

============================================================= ERRORS ==============================================================
_____________________________________________ ERROR collecting tests/test_launcher.py _____________________________________________
venv/lib/python3.14/site-packages/starlette/testclient.py:38: in <module>
    import httpx
E   ModuleNotFoundError: No module named 'httpx'

During handling of the above exception, another exception occurred:
tests/test_launcher.py:26: in <module>
    from fastapi.testclient import TestClient
venv/lib/python3.14/site-packages/fastapi/testclient.py:1: in <module>
    from starlette.testclient import TestClient as TestClient  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
venv/lib/python3.14/site-packages/starlette/testclient.py:40: in <module>
    raise RuntimeError(
E   RuntimeError: The starlette.testclient module requires the httpx package to be installed.
E   You can install this with:
E       $ pip install httpx
===================================================== short test summary info =====================================================
ERROR tests/test_launcher.py - RuntimeError: The starlette.testclient module requires the httpx package to be installed.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================== 1 error in 0.41s =========================================================
(venv) bash-5.3$ 
(venv) bash-5.3$ pip install httpx
Collecting httpx
  Using cached httpx-0.28.1-py3-none-any.whl.metadata (7.1 kB)
Requirement already satisfied: anyio in ./venv/lib/python3.14/site-packages (from httpx) (4.12.1)
Requirement already satisfied: certifi in ./venv/lib/python3.14/site-packages (from httpx) (2026.2.25)
Collecting httpcore==1.* (from httpx)
  Using cached httpcore-1.0.9-py3-none-any.whl.metadata (21 kB)
Requirement already satisfied: idna in ./venv/lib/python3.14/site-packages (from httpx) (3.11)
Requirement already satisfied: h11>=0.16 in ./venv/lib/python3.14/site-packages (from httpcore==1.*->httpx) (0.16.0)
Using cached httpx-0.28.1-py3-none-any.whl (73 kB)
Using cached httpcore-1.0.9-py3-none-any.whl (78 kB)
Installing collected packages: httpcore, httpx
Successfully installed httpcore-1.0.9 httpx-0.28.1
(venv) bash-5.3$ 
(venv) bash-5.3$ 
(venv) bash-5.3$ python -m pytest tests/test_launcher.py -v
======================================================= test session starts =======================================================
platform darwin -- Python 3.14.3, pytest-9.0.2, pluggy-1.6.0 -- /Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/mspreitz/go/src/github.com/llm-d-incubation/llm-d-fast-model-actuation/inference_server/launcher
plugins: anyio-4.12.1
collected 73 items                                                                                                                

tests/test_launcher.py::TestVllmConfig::test_vllm_config_with_env_vars PASSED                                               [  1%]
tests/test_launcher.py::TestVllmConfig::test_vllm_config_without_env_vars PASSED                                            [  2%]
tests/test_launcher.py::TestVllmInstance::test_instance_creation PASSED                                                     [  4%]
tests/test_launcher.py::TestVllmInstance::test_instance_start PASSED                                                        [  5%]
tests/test_launcher.py::TestVllmInstance::test_instance_start_already_running PASSED                                        [  6%]
tests/test_launcher.py::TestVllmInstance::test_instance_stop PASSED                                                         [  8%]
tests/test_launcher.py::TestVllmInstance::test_instance_stop_not_running PASSED                                             [  9%]
tests/test_launcher.py::TestVllmInstance::test_instance_force_kill PASSED                                                   [ 10%]
tests/test_launcher.py::TestVllmInstance::test_instance_get_status PASSED                                                   [ 12%]
tests/test_launcher.py::TestVllmInstance::test_instance_uuid_to_index_translation PASSED                                    [ 13%]
tests/test_launcher.py::TestVllmInstance::test_instance_uuid_translation_creates_env_vars_if_none PASSED                    [ 15%]
tests/test_launcher.py::TestVllmInstance::test_instance_uuid_translation_preserves_existing_env_vars PASSED                 [ 16%]
tests/test_launcher.py::TestVllmInstance::test_instance_no_uuid_translation_when_gpu_uuids_none PASSED                      [ 17%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_create_instance_auto_id PASSED                                    [ 19%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_create_instance_custom_id PASSED                                  [ 20%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_create_instance_duplicate_id PASSED                               [ 21%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_stop_instance PASSED                                              [ 23%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_stop_nonexistent_instance PASSED                                  [ 24%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_stop_all_instances PASSED                                         [ 26%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_get_instance_status PASSED                                        [ 27%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_get_instance_status_nonexistent PASSED                            [ 28%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_get_all_instances_status PASSED                                   [ 30%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_list_instances PASSED                                             [ 31%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_get_instance_log_bytes PASSED                                     [ 32%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_get_instance_log_bytes_nonexistent PASSED                         [ 34%]
tests/test_launcher.py::TestVllmMultiProcessManager::test_get_instance_log_bytes_with_end PASSED                            [ 35%]
tests/test_launcher.py::TestAPIEndpoints::test_health_endpoint PASSED                                                       [ 36%]
tests/test_launcher.py::TestAPIEndpoints::test_index_endpoint PASSED                                                        [ 38%]
tests/test_launcher.py::TestAPIEndpoints::test_create_vllm_instance PASSED                                                  [ 39%]
tests/test_launcher.py::TestAPIEndpoints::test_create_id_vllm_instance PASSED                                               [ 41%]
tests/test_launcher.py::TestAPIEndpoints::test_create_duplicate_instance PASSED                                             [ 42%]
tests/test_launcher.py::TestAPIEndpoints::test_delete_vllm_instance PASSED                                                  [ 43%]
tests/test_launcher.py::TestAPIEndpoints::test_delete_nonexistent_instance PASSED                                           [ 45%]
tests/test_launcher.py::TestAPIEndpoints::test_delete_all_instances PASSED                                                  [ 46%]
tests/test_launcher.py::TestAPIEndpoints::test_list_instances PASSED                                                        [ 47%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_status PASSED                                                   [ 49%]
tests/test_launcher.py::TestAPIEndpoints::test_get_nonexistent_instance_status PASSED                                       [ 50%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_endpoint PASSED                                            [ 52%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_with_range_header PASSED                                   [ 53%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_nonexistent_endpoint PASSED                                [ 54%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_range_not_available PASSED                                 [ 56%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_partial_content_206 PASSED                                 [ 57%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_open_ended_range PASSED                                    [ 58%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_invalid_range PASSED                                       [ 60%]
tests/test_launcher.py::TestAPIEndpoints::test_get_instance_logs_suffix_range_rejected PASSED                               [ 61%]
tests/test_launcher.py::TestVllmInstanceLogs::test_get_log_bytes_no_file PASSED                                             [ 63%]
tests/test_launcher.py::TestVllmInstanceLogs::test_get_log_bytes_empty_file PASSED                                          [ 64%]
tests/test_launcher.py::TestVllmInstanceLogs::test_get_log_bytes_with_content PASSED                                        [ 65%]
tests/test_launcher.py::TestVllmInstanceLogs::test_get_log_bytes_with_end PASSED                                            [ 67%]
tests/test_launcher.py::TestVllmInstanceLogs::test_get_log_bytes_persists_across_calls PASSED                               [ 68%]
tests/test_launcher.py::TestHelperFunctions::test_set_env_vars PASSED                                                       [ 69%]
tests/test_launcher.py::TestHelperFunctions::test_max_log_response_bytes_constant PASSED                                    [ 71%]
tests/test_launcher.py::TestLogFile::test_basic_retrieval PASSED                                                            [ 72%]
tests/test_launcher.py::TestLogFile::test_empty_file PASSED                                                                 [ 73%]
tests/test_launcher.py::TestLogFile::test_end_limit_enforcement PASSED                                                      [ 75%]
tests/test_launcher.py::TestLogFile::test_end_larger_than_content PASSED                                                    [ 76%]
tests/test_launcher.py::TestLogFile::test_unicode_bytes PASSED                                                              [ 78%]
tests/test_launcher.py::TestLogFile::test_partial_retrieval_with_end PASSED                                                 [ 79%]
tests/test_launcher.py::TestLogFile::test_start_offset PASSED                                                               [ 80%]
tests/test_launcher.py::TestLogFile::test_start_at_message_boundary PASSED                                                  [ 82%]
tests/test_launcher.py::TestLogFile::test_start_beyond_available_raises_error PASSED                                        [ 83%]
tests/test_launcher.py::TestLogFile::test_start_equal_to_length_raises_error PASSED                                         [ 84%]
tests/test_launcher.py::TestLogFile::test_no_file_with_start_raises_error PASSED                                            [ 86%]
tests/test_launcher.py::TestLogFileCleanup::test_stop_terminated_cleans_up_log_file PASSED                                  [ 87%]
tests/test_launcher.py::TestLogFileCleanup::test_stop_not_running_cleans_up_log_file PASSED                                 [ 89%]
tests/test_launcher.py::TestLogFileCleanup::test_cleanup_missing_file_no_error PASSED                                       [ 90%]
tests/test_launcher.py::TestParseRangeHeader::test_full_range PASSED                                                        [ 91%]
tests/test_launcher.py::TestParseRangeHeader::test_open_ended_range PASSED                                                  [ 93%]
tests/test_launcher.py::TestParseRangeHeader::test_large_values PASSED                                                      [ 94%]
tests/test_launcher.py::TestParseRangeHeader::test_suffix_range_rejected PASSED                                             [ 95%]
tests/test_launcher.py::TestParseRangeHeader::test_invalid_unit PASSED                                                      [ 97%]
tests/test_launcher.py::TestParseRangeHeader::test_garbage_input PASSED                                                     [ 98%]
tests/test_launcher.py::TestParseRangeHeader::test_end_less_than_start PASSED                                               [100%]

======================================================= 73 passed in 1.61s ========================================================
(venv) bash-5.3$ 
```


### Version

main (please specify commit below)

### Commit SHA

cf92c6d14cff3149e12b86b184cf81ceef0bad8a

### Relevant log output

```shell

```

--- a/.github/workflows/python-code-quality.yml
+++ b/.github/workflows/python-code-quality.yml
@@ -34,7 +34,7 @@ jobs:
           # install black isort flake
           pip install black==24.4.2 isort==5.13.2 flake8==7.0.0
           # install pytest and other dependencies
-          pip install pytest==9.0.2 fastapi==0.116.1 pydantic==2.11.7 uvicorn==0.35.0 uvloop==0.21.0 httpx==0.28.1 nvidia-ml-py==13.590.48 kubernetes==31.0.0
+          pip install pytest==9.0.2 -r inference_server/launcher/requirements-sans-vllm.txt
 
       - name: Check for trailing whitespace
         run: |

--- a/inference_server/launcher/howto.md
+++ b/inference_server/launcher/howto.md
@@ -1,13 +1,14 @@
 1- HOW TO RUN THE UNIT TEST:
 
-Install all the necessary packages:
+Install all the necessary packages (feel free to use a later version of pytest if you prefer):
 ```bash
-pip install -r requirements.txt
+pip install pytest==9.0.2 -r requirements-sans-vllm.txt
 ```
 
 Run the unit test doing:
 ```bash
-python -m pytest test_launcher.py -v
+python -m pytest tests/test_launcher.py -v
+python -m pytest tests/test_gputranslator.py -v
 ```
 
 2- RUN E2E TEST:

--- a/inference_server/launcher/requirements-sans-vllm.txt
+++ b/inference_server/launcher/requirements-sans-vllm.txt
@@ -1,0 +1,7 @@
+fastapi==0.135.1
+pydantic==2.12.5
+uvicorn==0.42.0
+uvloop==0.22.1
+httpx==0.28.1
+nvidia-ml-py==13.590.48
+kubernetes==31.0.0

--- a/inference_server/launcher/requirements.txt
+++ b/inference_server/launcher/requirements.txt
@@ -1,8 +1,3 @@
-fastapi
-pydantic
-uvicorn
-uvloop
-nvidia-ml-py
-kubernetes==31.0.0
+-r requirements-sans-vllm.txt
 # WARNING: vllm must be built from source on a macOS Silicon
 vllm; sys_platform != "darwin" or platform_machine != "arm64"

